packages/data-sdk/src/Device.ts

```diff
@@ -262,38 +263,63 @@ export class Device extends EventEmitter implements IRealtimeDevice {
         }
       }
 
-      // Each online device and user has a peer in the system
-      const peers = await rtcClient.getPeers();
+      // WebRTC requires a signaling phase when forming a new connection.
 
-      // Find the device peer corresponding to the device's ID
-      const devicePeer = peers.find((_) => _.deviceId === this.id);
-      if (!devicePeer) {
-        // If the device is offline, we won't be able to find its peer.
-        throw new Error("Cannot find peer, is the robot offline?");
-      }
+      this.remoteDevicePeerId = await this.getRemoteDevicePeerId(rtcClient);
 
-      // We can connect our real-time communication client to device peers by their ID
-      this.remoteDevicePeerId = devicePeer.id;
-      await (rtcClient as RtcClient).connect(this.remoteDevicePeerId);
+      const sessionId = await this.createSession(rtcClient);
 
-      // WebRTC requires a signaling phase when forming a new connection.
       // Wait for the signaling process to complete...
-      while (
-        rtcClient.getConnectionStatus(this.remoteDevicePeerId) !== "connected"
-      ) {
-        await delay(100);
-      }
-      this.rtcClient = rtcClient;
-      this.emit("connect");
+      if (!!sessionId) {
+        while (
```

Comment:
Wouldn't it still be possible to get stuck in this loop? I.e. maybe the internet suddenly drops and now this RTC session will never connect.